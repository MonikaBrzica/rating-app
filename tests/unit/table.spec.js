import { shallowMount, createLocalVue } from '@vue/test-utils'
import Table from '../../src/components/table'
import Vuex from 'vuex'

const localVue = createLocalVue()
localVue.use(Vuex)
const store = new Vuex.Store({
  actions: {
  },
  state: {
      ratings: [
          {
              emojiId: 1,
              date: '2021-08-16T06:00:00Z'
          },
          {
            emojiId: 2,
            date: '2021-08-16T07:00:00Z'
        }
      ]
  },
  getters: {
    getSumRatings (state) {
        const ratings = state.ratings
        const series = [0, 0, 0, 0, 0]
        if (ratings !== null) {
          for (let i = 0; i < ratings.length; i++) {
            if (ratings[i].emojiId === 1) {
              series[0]++
            }
            if (ratings[i].emojiId === 2) {
              series[1]++
            }
            if (ratings[i].emojiId === 3) {
              series[2]++
            }
            if (ratings[i].emojiId === 4) {
              series[3]++
            }
            if (ratings[i].emojiId === 5) {
              series[4]++
            }
          }
        }
        return series
      }
  }
})

describe('table', () => {
  const wrapper = shallowMount(Table, {
    store,
    localVue
  })
  it('renders correctly', () => {
    expect(wrapper.element).toMatchSnapshot()
  })
})
