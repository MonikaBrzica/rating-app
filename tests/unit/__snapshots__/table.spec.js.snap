// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`table renders correctly 1`] = `
<div
  class="table"
>
  <table>
    <thead>
      <tr>
        <th>
          Emoticon
        
          <img
            alt="drop-down"
            src="../assets/images/caret-down-solid.svg"
          />
        </th>
         
        <th>
          Count
        
          <img
            alt="drop-down"
            src="../assets/images/caret-down-solid.svg"
          />
        </th>
      </tr>
    </thead>
     
    <tbody>
      <tr>
        <td>
          Very happy
        </td>
         
        <td>
          1
        </td>
      </tr>
      <tr>
        <td>
          Happy
        </td>
         
        <td>
          1
        </td>
      </tr>
      <tr
        style="display: none;"
      >
        <td>
          Meh
        </td>
         
        <td>
          0
        </td>
      </tr>
      <tr
        style="display: none;"
      >
        <td>
          Sad
        </td>
         
        <td>
          0
        </td>
      </tr>
      <tr
        style="display: none;"
      >
        <td>
          Very sad
        </td>
         
        <td>
          0
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;
